### DISCUSSION TITLE
Self hosting on linode the docker of outline just dies out

### DISCUSSION THREAD

**Discussion by sanjaygeeky** [outline/outline#5498] [Self Hosting]
Hi
I am trying to host the outline on linode ubuntu 22.04 All ports opened from the link
https://docs.getoutline.com/s/hosting/doc/docker-7pfeLP5a8t
version: "3"
services:

  outline:
    image: docker.getoutline.com/outlinewiki/outline:latest
    command: sh -c "yarn --env=production-ssl-disabled && yarn start --env=production-ssl-disabled"
    ports:
      - "3000:3000"
    depends_on:
      - postgres
      - redis
 
  redis:
    image: redis:7
    restart: always
    env_file: docker.env
    expose:
      - "6379"
    volumes:
      - ./redis.conf:/redis.conf
    command: ["redis-server", "/redis.conf"]
 
  postgres:
    image: postgres:14
    restart: always
    env_file: docker.env
    expose:
      - "5432"
    volumes:
      - database-data:/var/lib/postgresql/data
 
volumes:
  database-data:

    env_file: ./docker.env
    ports:
      - "3000:3000"
    depends_on:
      - postgres
      - redis
      - storage

  redis:
    image: redis
    env_file: ./docker.env
    ports:
      - "6379:6379"
    volumes:
      - ./redis.conf:/redis.conf
    command: ["redis-server", "/redis.conf"]
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 10s
      timeout: 30s
      retries: 3

  postgres:
    image: postgres
    env_file: ./docker.env
    ports:
      - "5432:5432"
    volumes:
      - database-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD", "pg_isready"]
      interval: 30s
      timeout: 20s
      retries: 3
    environment:
      POSTGRES_USER: 'user'
      POSTGRES_PASSWORD: 'pass'
      POSTGRES_DB: 'outline'

  storage:
    image: minio/minio
    env_file: ./docker.env
    ports:
      - "9000:9000"
    entrypoint: sh
    command: -c 'minio server'
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - storage-data:/data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

volumes:
  storage-data:
  database-data:

and ./docker.env is 
NODE_ENV=production
SECRET_KEY=secretkey
UTILS_SECRET=utilssecret

#postgres
DATABASE_URL=postgres://user:pass@IPV4:5432/outline
DATABASE_URL_TEST=postgres://user:pass@IPV4:5432/outline-test
DATABASE_CONNECTION_POOL_MIN=
DATABASE_CONNECTION_POOL_MAX=
# Uncomment this to disable SSL for connecting to Postgres
PGSSLMODE=disable

#redis
REDIS_URL=redis://IPV4:6379

URL=http://IPV4:3000
PORT=3000

AWS_ACCESS_KEY_ID=KEY_ID
AWS_SECRET_ACCESS_KEY=ACCESS_KEY
AWS_REGION=ap-south-1
AWS_S3_ACCELERATE_URL=
AWS_S3_UPLOAD_BUCKET_URL=http://s3:4569
AWS_S3_UPLOAD_BUCKET_NAME=bucketname
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private
#IAM outlinebucketaccess policy

#Authentication
# When configuring the Client ID, add an Authorized redirect URI:
# https://outline.company.com/auth/google.callback
GOOGLE_CLIENT_ID=13XXXXXXXXXXXXXXXXXXapps.googleusercontent.com
GOOGLE_CLIENT_SECRET=GXXXXXXXXXXXXXXXXXx

Am I missing something

**Comment by heyvaldemar** [comment]
hey guys, I wrote [a detailed guide](https://www.heyvaldemar.com/install-outline-and-keycloak-using-docker-compose/) after I was able to configure Outline with MinIO and Keycloak.

Hopefully, it will help others.
